kedro-datasets/kedro_datasets/ibis/table_dataset.py

```diff
@@ -166,7 +178,11 @@ def load(self) -> ir.Table:
             reader = getattr(self.connection, f"read_{self._file_format}")
             return reader(self._filepath, self._table_name, **self._load_args)
         else:
-            return self.connection.table(self._table_name)
+            return (
+                self.connection.table(self._table_name)
+                if self._database is None
+                else self.connection.table(self._table_name, **self._load_args)
+            )
```

Comment:
```suggestion
            return self.connection.table(self._table_name, **self._load_args)
```
This seems unnecessary

Reply:
This was to address backends that don't support `database` (e.g. polars), we discussed [here](https://kedro-org.slack.com/archives/D06RN67G5UY/p1731624487233569?thread_ts=1731624136.981659&cid=D06RN67G5UY)

Reply:
> Ah, right! I think you set it up in `__init__()` though such that `database` is only added to `self._load_args` if it's not `None`, so this simplification is probably still correct. Maybe it makes sense to add a Polars test if making this change, just to be safe, though.

Agree doing it in `_init_()` would make sense :+1:  I recall looking to do it there first, but it felt more complicated because of this [reader](https://github.com/kedro-org/kedro-plugins/blob/9770b481db64cd3597d9ab01e1db674c478d2d63/kedro-datasets/kedro_datasets/ibis/table_dataset.py#L178-L179) code which I think is what's going to get deprecated in favor of `FileDataset`? The `reader` fails if `load_args` has a `database` parameter so I can't be passed in those use cases... 

I could modify the `_init_()` to pass `database` to `load_args` `if self._database is not None and self._filepath is None:` if that seems more readable?

Reply:
> I also am not 100% sure if it could be an issue that `**self._load_args` isn't being included if `database is None`, in case something beyond `name` and `database` can be passed to `Backend.table()`; asking here: https://ibis-project.zulipchat.com/#narrow/channel/405263-general/topic/Are.20there.20any.20restrictions.20around.20.60Backend.2Etable.60.20signature.3F/near/486157200
> 
> (Also probably relevant to some of the questions from @ravi-kumar-pilla in #956, although the use of `**self._load_args` here is one of the reasons why it shouldn't be removed in that PR regardless.)

Definitely makes sense to confirm, but I did review several ibis backends and test the few I use to confirm that database defaults to `None` for what it's worth